railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -0,0 +1,253 @@
+package ctrlutils
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/deck/file"
+	"github.com/kong/kubernetes-ingress-controller/pkg/sendconfig"
+	"github.com/prometheus/common/log"
+
+	"sync"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	knative "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	knativeversioned "knative.dev/networking/pkg/client/clientset/versioned"
+	knativeApis "knative.dev/pkg/apis"
+	"knative.dev/pkg/network"
+)
+
+func PullConfigUpdate(kongConfig sendconfig.Kong, log logr.Logger, ctx context.Context, kubeConfig *rest.Config, stopCh <-chan struct{}) {
+	log.Info("Launching Customer Resource Update thread.")
+	var wg sync.WaitGroup
+	for {
+		select {
+		case updateDone := <-kongConfig.ConfigDone:
+			log.Info("receive configuration information. Update ingress status \n%v\n \n", &updateDone)
+			wg.Add(1)
+			go UpdateIngress(&updateDone, log, ctx, kubeConfig, &wg)
+		case <-stopCh:
+			log.Info("stop status update channel.")
+			return
+		}
+	}
+}
+
+// update ingress status according to generated rules and specs
+func UpdateIngress(targetContent *file.Content, log logr.Logger, ctx context.Context, kubeconfig *rest.Config, wg *sync.WaitGroup) error {
+	defer wg.Done()
+	for _, svc := range targetContent.Services {
+		for _, plugin := range svc.Plugins {
+			log.Info("\n service host %s name %s plugin enablement %v\n", *svc.Service.Host, *svc.Service.Name, *svc.Plugins[0].Enabled)
+			if *plugin.Enabled == true {
+				// filter the plugins (tcp/udp/tls/http) here
+				if config, ok := plugin.Config["add"]; ok {
```

Comment:
What's the rationale for the plugin-related conditions here? I'm not sure why we'd expect a particular configuration of a specific plugin (which isn't indicated, but the specific config key indicates this is looking for a particular plugin's configuration).

Reply:
Shared the steps running the test cases locally
and displayed the expected output for travis. 
The concern is sorted out in slack.